// multicopter status and command message

syntax = "proto2";

message PbVec3 {
  required float x = 1;
  required float y = 2;
  required float z = 3;
}

message PbVec4 {
  required float x = 1;
  required float y = 2;
  required float z = 3;
  required float w = 4;
}

message PbCopterState {
    optional PbVec3 accl = 1;
    optional PbVec3 gyro = 2;
    optional PbVec3 mag = 3;

    optional PbVec4 controls = 4;

    optional PbVec4 refOrientation = 5;

    optional PbVec4 orientation = 6;

    optional PbVec4 motorSpeed = 7;
}

message PbCopterCommand {
    optional PbVec3 rollPidParams = 1;
    optional PbVec3 pitchPidParams = 2;
    optional PbVec3 yawPidParams = 3;
}
